pymodbus/server/async_io.py

```diff
@@ -173,17 +175,28 @@ def execute(self, request, *addr):
         if self.server.request_tracer:
             self.server.request_tracer(request, *addr)
 
+        task = self.loop.create_task(self._async_execute(request, *addr))
```

Comment:
This is a lot of overhead, creating a task for every request. It is possible and efficient to call async methods from sync methods.

Reply:
Seems that asyncio documentation seems to prefer create_task, but if you have proposal to alternative approach, I can change it.

Reply:
I changed this to run_coroutine_threadsafe, ensure_future seemed to end up using create_task.